pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -161,9 +164,9 @@ func (h *handler) OnJobChange(_ string, job *batchv1.Job) (*batchv1.Job, error)
 		return job, err
 	}
 
-	infraObj, err := newInfraObject(infraMachine)
+	infra, err := newInfraObject(infraMachine)
 	if err != nil {
-		return nil, err
+		return job, err
```

Comment:
@Oats87 IIRC if `err != nil`, the `OnChange` handler will discard the returned object, so this is merely a stylistic choice.